### ISSUE TITLE
Reproducing results for QM9

### ISSUE THREAD

**Issue by shehzaidi** [torchmd/torchmd-net#64]
Hi! I have recently been trying to reproduce your results for QM9 and have some questions. I forked your code this week and started a run on QM9 for predicting the HOMO energy. I only changed `dataset-arg` and `log-dir`, otherwise everything is identical to the latest version of your code I think. The command I used is:

`CUDA_VISIBLE_DEVICES=0,1 python scripts/train.py --conf examples/ET-QM9.yaml --dataset QM9 --log-dir output/ --dataset-arg homo`

I matched the GPU type and count to your paper: I launched over two RTX 2080 Ti's.

However, I only seem to achieve 26 meV for the HOMO test MAE. Do you have any suggestions for what might be differing in my setup compared to yours? Do I need to change any of the other configs in `examples/ET-QM9.yaml` to train on HOMO? How much variance should I expect across seeds?

Thank you for your help! :slightly_smiling_face: 

Sheh

**Comment by PhilippThoelke** [comment]
Hi, thank you for the interest in reproducing the results! The HOMO label does not have atomic reference values but the `examples/ET-QM9.yaml` defines `Atomref` as the prior model. You can replace this with `null` to not use any priors. We also set `standardize` to true for predicting HOMO, which helps the model with convergence. The model's output `y` is multiplied with the training set's standard deviation and added to its mean as `y * std + mean`.

Let me know if this resolves the problem!

**Comment by PhilippThoelke** [comment]
Closing this for now, feel free to reopen if the problem was not resolved.

**Comment by shehzaidi** [comment]
Thanks, Philipp! After making those two changes, I do get performance in the right ballpark :slightly_smiling_face: 

For the other targets in QM9, when do you use standardization and the prior model? Just wondering what the precise configs need to be to reproduce the results on all QM9 targets.

**Comment by PhilippThoelke** [comment]
Perfect! The idea is to make the magnitude of output values "easier" for the network by centering around 0 and having a reasonably small standard deviation. For those targets in QM9 that have atomic reference values, this already does the job and it's even more powerful than our standardization as we can basically have the mean target for each atom type. Therefore, wherever there is an atom ref in QM9, we use that instead of standardizing manually. For the dipole moments and electronic spatial extent, we are using specialized output networks, which do not require standardization. For all other targets (those with default scalar output networks and no atom refs) we use standardization.

That makes:
- atom ref prior: energy_U0, energy_U, enthalpy_H, free_energy
- standardization: isotropic_polarizability, homo, lumo, gap, zpve, heat_capacity
- neither (but special output network): dipole_moment, electronic_spatial_extent

You can set the output network  by setting `output_model` to something other than `Scalar`. Currently implemented are the following output modules: `Scalar`, `DipoleMoment`, `ElectronicSpatialExtent` as also indicated when you run `python scripts/train.py --help`.
